src/components/forms/AttributeForm.jsx

```diff
@@ -0,0 +1,125 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { useTranslation } from 'react-i18next';
+import StringField from '../input/StringField';
+import BoolField from '../input/BoolField';
+import DateTimeField from '../input/DateTimeField';
+import IntegerField from '../input/IntField';
+import FloatField from '../input/FloatField';
+import { IconButton, InputAdornment, Tooltip } from '@material-ui/core';
+import { Clear as ClearIcon } from '@material-ui/icons';
+import Key from '@material-ui/icons/VpnKey';
+import Typography from '@material-ui/core/Typography';
+const useStyles = makeStyles(() => ({
+  header: {
+    marginTop: '1rem',
+    marginLeft: '2rem',
+  },
+  field: {
+    marginLeft: '5rem',
+    marginRight: '5rem',
+    minWidth: '10rem',
+    maxWidth: '40rem',
+  },
+}));
+
+const field = {
+  String: StringField,
+  Boolean: BoolField,
+  DateTime: DateTimeField,
+  Integer: IntegerField,
+  Float: FloatField,
+};
+
+export default function AttributeForm({ records, setRecords, ...props }) {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const completedAttr = () => {
+    let number = 0;
+    for (let attrName of Object.keys(records)) {
+      if (records[attrName] !== null && records[attrName] !== undefined) {
+        number += 1;
+      }
+    }
+    return number;
+  };
+
+  const handleSetValue = (attrName) => (value) => {
+    setRecords({ ...records, [attrName]: value });
+  };
```

Comment:
I like this simplification of the `handleSetValue` function as a HOF. However, I believe the `setRecords` prop could be changed to `onChange` and executed as a callback, rather than directly modifying the state of the parent.

The function signature could be like this:

```ts
function onChange (attrName: string, value: AttributeValueType) => void;
// alternatively
function onChange ({ attrName: string, value: AttributeValueType }) => void;
```

This assumes less knowledge of the parent by the `AttributeForm` component. It only exposes an `onChange` callback that the parent uses as it sees fit.

To be considered for the future: we will need to consider other submit options for the form, rather than updating parent values every time an input changes.